JS.md

```diff
@@ -0,0 +1,233 @@
+ * [[WasmInstance]] : the [`Eval.instance`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L15)
+   containing the exported function
+ * [[WasmFunctionIndex]] : the index of the function inside the module
+
+Exported Functions also have the following data properties:
+* the `length` property is set to the exported function's signature's arity 
+* the `name` is set to `index` as a Number value
+
+WebAssembly Exported Functions have a `[[Call]](this, argValues)` method defined as:
+ * Let `argTypes` be the list of value types defined by the signature of [[WasmFunctionIndex]].
+ * Let `args` be an empty list of coerced values.
+ * For each value type `t` in `argTypes` and value `v` in `argValues`:
+   * Append to `args` `v` coerced to `t` as follows:
+     * coerce `v` to `i32` via [`ToInt32(v)`](http://tc39.github.io/ecma262/#sec-toint32)
+     * throw a `TypeError` if `t` is `i64`
+     * coerce `v` to `f32` by first applying [`ToNumber(v)`](http://tc39.github.io/ecma262/#sec-tonumber)
+       and then converting the resulting IEEE754 64-bit double to a 32-bit float using `roundTiesToEven`
+     * coerce `v` to `f64` via [`ToNumber(v)`](http://tc39.github.io/ecma262/#sec-tonumber)
+ * Perform [`Eval.invoke`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L327)
+   passing [[WasmInstance]], [[WasmFunctionIndex]], and `args`.
+ * Coerce the result of `Eval.invoke` as follows:
+   * if the return value is `None`, return `undefined`
+   * interpret `i32` as a signed integer and convert that integer to a Number value
+   * throw a `TypeError` if returning an `i64`
+   * return `f32`/`f64` as Number values, possibly performing
+     [canonicalization of NaNs](http://tc39.github.io/ecma262/#sec-setvalueinbuffer)
+
+Exported Functions do not have a [[Construct]] method and thus it is not possible to 
+call one with the `new` operator.
+
+## `Wasm.Instance` Objects
+
+A `Wasm.Instance` object represents the activation/instantiation of a single
+`Wasm.Module` into a single [realm](http://tc39.github.io/ecma262/#sec-code-realms)
+with a tuple of realm-specific JS import values. A `Wasm.Instance` logically
+contains a single [`Eval.instance`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L15).
+
+### `Wasm.Instance` Constructor
+
+Attempting to call `Wasm.Instance` as a function or constructor directly from JS
+throws a `TypeError`. `Wasm.Instance` instances can only be created indirectly from
+JS by calling [`instantiate`](JS.md#wasmmoduleprototypeinstantiate).
+
+### `Wasm.Instance` `exports` property
+
+`Wasm.Instance` objects are created with an initial (writable, configurable,
+non-enumerable) `exports` data property whose value is the JS Array of exported
+JS Functions created by [`instantiate`](JS.md#wasmmoduleprototypeinstantiate).
+
+## Sample API Usage
+
+Given `demo.was` (encoded to `demo.wasm`):
```

Comment:
I'm trying to be more regular about using `.was` for "a single wasm module in the text format" and `.wast` for "the webassembly test file format, which can contain multiple modules as well as tests run on those modules".  In this case, I mean `.was` since we're talking about a single module.